compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorLowering.cpp

```diff
@@ -101,14 +101,10 @@ void LLVMCPUVectorLoweringPass::runOnOperation() {
 
   // Make sure we remove redundant vector ops (e.g., vector tranposes) before we
   // lower them and can't be optimized away anymore.
+  // TODO (dcaballe): We should run full canonicalization here.
```

Comment:
should we? it looks like we are running less canonicalization patterns here

Reply:
Unfortunately, the previous code is not doing what we thought. It applies dialect-level canonicalization patterns, not all the canonicalization patterns from all the ops in that dialect. We went from running full canonicalization and cse between these passes to not running anything, which is leading to unexpected patterns not being applied and other issues...